packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -540,17 +625,14 @@ export function makeVirtualObjectManager(
         propertyNames.add(prop);
       }
       const innerSelf = { vobjID, rawData, repCount: 0 };
-      const [initialRepresentative, wrappedData] = makeRepresentative(
-        innerSelf,
-        true,
-        false,
-      );
-      registerEntry(vobjID, initialRepresentative);
+      const [representative, exposedRepresentative, wrappedData] =
```

Comment:
So `representative` here is the "cohort record": it's the thing that gets registered in the FinalizationRegistry, so it's what GC will be watching for. What keeps it alive?

I see the `stateToRepresentative.set(wrappedData, storedRepresentative)`, and `wrappedData` is really the `state` object that is passed to the user's `actualize` and `finish` functions, and `storedRepresentative` is returned by `makeRepresentative()` and becomes this `representative` value. (complaint: having the same value take on different names in different places makes it really hard to trace this through).

There's nothing else here which keeps `wrappedData`/`state` alive. So if any of the user's facets don't happen to capture `state`, then `state` might get dropped even though one of the facets is still alive, which means the cohort record will get dropped, which means `bringOutYourDead` will think the entire cohort has died, when in fact the in-memory pillar should still be standing.

I'm bothered that this isn't caught by a unit test. I started to write one, was stymied by the VOM cache (even setting the size to 1, I had to allocate a bunch of dummy objects to get `state` to fall out of the cache and see the GC I was expecting), and stopped when my extra `console.log`s showed me the problem was real. In case it helps, the setup looks like this:

```js
import { Far } from '@endo/marshal';
const { defineKind } = VatData;
export function buildRootObject(vatPowers) {
  const makeFoo = defineKind(
    'foo',
    () => ({ prop: 'value' }),
    state => ({
      facet1: { foo: () => 0 },
      facet2: { bar: () => 0 },
    }),
  );
  let facet1;
  return Far('root', {
    bootstrap(_vats) {},
    start() {
      facet1 = makeFoo().facet1;
      makeFoo();
      makeFoo();
      makeFoo();
    },
    drop() {
      facet1 = null;
    },
  });
}
```

and to drive it:
```js
// eslint-disable-next-line import/order
import { test } from '../tools/prepare-test-env-ava.js';

// eslint-disable-next-line import/order
import { buildVatController } from '../src/index.js';
import { capargs } from './util.js';

test('drop', async t => {
  const config = {
    defaultManagerType: 'xs-worker',
    bootstrap: 'vat1',
    vats: {
      vat1: {
        sourceSpec: new URL('vat-4742.js', import.meta.url).pathname,
        creationOptions: { virtualObjectCacheSize: 1 },
      },
    },
  };
  const c = await buildVatController(config);
  c.pinVatRoot('vat1');
  await c.run();
  console.log(`-- READY`);

  c.queueToVatRoot('vat1', 'start', capargs([]));
  await c.run();

  console.log(`-- dropping`);
  c.queueToVatRoot('vat1', 'drop', capargs([]));
  await c.run();
});
```

I think the fix needs a WeakMap from each Representative (facet) to the cohort record (named `representative` here). So every facet is involved in a reference cycle with every other facet, as well as the cohort record (which is what actually matters, since it's what the FinalizationRegistry is watching).